resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/SparkNomadJobController.scala

```diff
@@ -75,6 +77,31 @@ private[spark] class SparkNomadJobController(jobManipulator: NomadJobManipulator
     }
   }
 
+  def removeExecutors(executorIds: Seq[String]): Future[Boolean] = {
+    try {
+      executorIds
+        .map(f => f.replaceFirst("-\\d+$", ""))
+        .foreach(f => {
+          logInfo("stopping alloc " + f)
+          jobManipulator.stopAlloc(f)
+        })
+    } catch {
+      case e: Exception => logError("caught an exception removing executors", e)
+        return Future.successful(false)
+    }
+
+    val removeCount = executorIds.size
+    jobManipulator.updateJob(startIfNotYetRunning = false) { job =>
```

Comment:
updating the executor task group count to match based on the allocations we just removed